gemini/use-cases/retrieval-augmented-generation/utils/intro_multimodal_rag_utils.py

```diff
@@ -1,82 +1,66 @@
 import os
+from typing import Optional, Dict, Tuple, List, Union, Any, Iterable
 import time
-import fitz
-import errno
+import glob
 import typing
+from base64 import b64encode
 import requests
 import numpy as np
 import pandas as pd
-import seaborn as sns
-import networkx as nx
-from typing import Optional
-from base64 import b64encode
-from typing import Tuple, List
-from typing import Dict, List, Union, Any, Iterable
-import matplotlib.pyplot as plt
+import fitz
 from google.cloud import aiplatform
 from google.protobuf import struct_pb2
-from IPython.display import Markdown, display
+from IPython.display import display
 from vertexai.language_models import TextEmbeddingModel
 from vertexai.generative_models import (
-    GenerativeModel,
-    GenerationConfig,
     Image,
-    Content,
-    Part,
-    GenerationResponse,
+    HarmCategory,
+    HarmBlockThreshold,
 )
 import PIL
 
+
+text_embedding_model = TextEmbeddingModel.from_pretrained("textembedding-gecko@latest")
+
+# textembedding-gecko@latest [Always pulls the latest]
+# textembedding-gecko@003 [Latest as on 14Feb 2024]
+
 # Function for getting text and image embeddings
 
 
 def get_text_embedding_from_text_embedding_model(
```

Comment:
Is there a reason you're using the textembedding model instead of the multimodal embedding model?

Don't all of the embeddings need to be made with the same model for them to match correctly?